src/main/java/Player.java

```diff
@@ -38,24 +38,29 @@ public void movePlayer (Coordinate c) {
         coord = c;
     }
 
-    public void setName(String n) {this.name = n;}
-
     public Coordinate position() {return coord;}
 
     public void output(String s) { printer.output(s); }
 
+    public void addItem(Item i) { backpack.add(i); }
+
+    public final void removeItem(Item i) { backpack.remove(i); }
+
+    public boolean instructionLengthValid(){ return parser.instructionLengthValid(); }
+
+    public int getScore() {return moves/(MAX_HEALTH + 1 - health);}
+
     public String getLine() { return parser.getLine(); }
 
     public List<Item> getBackpack () {
         return backpack;
     }
 
-    public void addItem(Item i) { backpack.add(i); }
-
-    public final void removeItem(Item i) { backpack.remove(i); }
+    public String getLastValidDirection() { return lastValidDirection; }
 
-    public int getScore() {return moves/(MAX_HEALTH + 1 - health);}
+    public void setName(String n) {this.name = n;}
 
     public void setLastValidDirection(String d) {lastValidDirection = d;}
-    public String getLastValidDirection() {return lastValidDirection;}
+
+    public void setInstructionLengthValid(boolean instructionLengthValid){ parser.setInstructionLengthValid(instructionLengthValid); }
```

Comment:
I think this should be a constant the PlayerThread is told and the Instruction object does not know.